File: app-core/src/main/java/ru/micron/dto/omdb/FilmResponseDto.java
package ru.micron.dto.omdb;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class FilmResponseDto {

  @JsonProperty("Title")
  private String title;

  @JsonProperty("Year")
  private String year;

  @JsonProperty("Rated")
  private String ageRate;

  @JsonProperty("Released")
  private String releaseYear;

  @JsonProperty("Runtime")
  private String filmTime;

  @JsonProperty("Genre")
  private String genres;

  @JsonProperty("Director")
  private String director;

  @JsonProperty("Writer")
  private String writers;

  @JsonProperty("Actors")
  private String actors;

  @JsonProperty("Plot")
  private String plot;

  @JsonProperty("Language")
  private String language;

  @JsonProperty("Country")
  private String country;

  @JsonProperty("Awards")
  private String awards;

  @JsonProperty("Poster")
  private String posterUrl;

  @JsonProperty("Ratings")
  private List<RatingsDto> ratings;

  @JsonProperty("Metascore")
  private String metascore;

  private String imdbRating;
  private String imdbVotes;
  private String imdbID;

  @JsonProperty("Type")
  private String type;

  @JsonProperty("DVD")
  private String dvd;

  @JsonProperty("BoxOffice")
  private String boxOffice;

  @JsonProperty("Production")
  private String production;

  @JsonProperty("Website")
  private String website;
}

File: app-core/src/main/kotlin/ru/micron/dto/omdb/FilmResponseDto.kt
package ru.micron.dto.omdb

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class FilmResponseDto(
    @JsonProperty("Title")
    var title: String? = null,
    @JsonProperty("Year")
    var year: String? = null,
    @JsonProperty("Rated")
    var ageRate: String? = null,
    @JsonProperty("Released")
    var releaseYear: String? = null,
    @JsonProperty("Runtime")
    var filmTime: String? = null,
    @JsonProperty("Genre")
    var genres: String? = null,
    @JsonProperty("Director")
    var director: String? = null,
    @JsonProperty("Writer")
    var writers: String? = null,
    @JsonProperty("Actors")
    var actors: String? = null,
    @JsonProperty("Plot")
    var plot: String? = null,
    @JsonProperty("Language")
    var language: String? = null,
    @JsonProperty("Country")
    var country: String? = null,
    @JsonProperty("Awards")
    var awards: String? = null,
    @JsonProperty("Poster")
    var posterUrl: String? = null,
    @JsonProperty("Ratings")
    var ratings: MutableList<RatingsDto>? = null,
    @JsonProperty("Metascore")
    var metascore: String? = null,
    var imdbRating: String? = null,
    var imdbVotes: String? = null,
    var imdbID: String? = null,
    @JsonProperty("Type")
    var type: String? = null,
    @JsonProperty("DVD")
    var dvd: String? = null,
    @JsonProperty("BoxOffice")
    var boxOffice: String? = null,
    @JsonProperty("Production")
    var production: String? = null,
    @JsonProperty("Website")
    var website: String? = null
)
